### DISCUSSION TITLE
Create Barrier.pem also with cmdline barriers script

### DISCUSSION THREAD

**Discussion by ghost** [input-leap/input-leap#1779] [Feature Requests]
> This issue has been migrated from old Barrier Github repository https://github.com/debauchee/barrier/issues/965
> 
> Issue created on: 2020-12-03 by @wummel
> Issue last updated on: 2020-12-07

After starting barriers 2.3.3 on Debian Linux with --enable-crypt I get this error:
```
/usr/bin/barriers --enable-crypto --no-daemon --address 192.168.178.2 --config ~/.config/barrier/barrier.conf
started server (IPv4), waiting for clients
[2020-12-03T16:10:01] INFO: OpenSSL 1.1.1d  10 Sep 2019
[2020-12-03T16:10:01] ERROR: ssl certificate doesn't exist: $HOME/.local/share/barrier/SSL/Barrier.pem
[2020-12-03T16:10:05] NOTE: stopped server
```
But after starting the GUI "barrier", Barrier.pem is automatically generated. I detected this feature after searching the source code.

I expected Barrier.pem to be generated when starting barriers --enable-crypto.
Also, it would be nice to see some documentation about the automatic .pem-file generation in the .man pages.

thanks

---

> Commented on: 2020-12-06 by @albertony

For information: I did actually include some documentation for how to generate the certificate and fingerprint manually in my recent wiki pull request https://github.com/debauchee/barrier-wiki/pull/5. See https://github.com/albertony/barrier-wiki/blob/local_config/Command-Line.md.

I included an example for Windows only, but I think for linux something like this should be equivalent:
```
openssl req -x509 -nodes -days 365 -subj /CN=Barrier -newkey rsa:2048 -keyout ~/.local/share/barrier/SSL/Barrier.pem -out ~/.local/share/barrier/SSL/Barrier.pem
openssl x509 -fingerprint -sha1 -noout -in ~/.local/share/barrier/SSL/Barrier.pem > ~/.local/share/barrier/SSL/SSL/Fingerprints\Local.txt
```

If your os setup uses XDG_DATA_HOME, i think the path prefix `~/.local/share/barrier/SSL` should be replaced with `$XDG_DATA_HOME/barrier/SSL`.

Found the openssl arguments to use from https://github.com/debauchee/barrier/blob/master/src/gui/src/SslCertificate.cpp.

**Comment by Alchemyst0x** [comment]
Not that this is much different, aside from the backslash/forward slash mix-up, just wanted to confirm this worked for me on Linux and I even saved it as a .sh file for ease-of-use :) Code below. Thanks for the quick fix/help & Cheers!

```bash
#!/usr/bin/env bash

openssl req -x509 -nodes -days 365 -subj /CN=Barrier \
	-newkey rsa:2048 \
	-keyout "$HOME/.local/share/barrier/SSL/Barrier.pem" \
	-out 	"$HOME/.local/share/barrier/SSL/Barrier.pem" &&

mkdir -vp "$HOME/.local/share/barrier/SSL/SSL/Fingerprints" &&

openssl x509 -fingerprint -sha1 -noout -in \
	"$HOME/.local/share/barrier/SSL/Barrier.pem" > \
	"$HOME/.local/share/barrier/SSL/SSL/Fingerprints/Local.txt" &&

echo -e "OK."
```
